experimental/InjectiveResolutions/src/InjectiveResolutions.jl

```diff
@@ -0,0 +1,1098 @@
+  irreducible_comp = [quotient_ring_as_module(Ji) for Ji in irreducible_ideals]
+  return direct_sum(irreducible_comp...,task=:none)
+end
+struct IrrRes # irreducible resolution (including all computed data and the cochain complex)
+  mod::SubquoModule
+  irr_sums::Vector{InjMod}
+  cochain_maps::Vector{SubQuoHom}
+  cochain_complex::ComplexOfMorphisms # if sequence not exact return trivial cochain_complex (M0 -> M0)
+end
+
+#direct sum of two injective modules over the same monoid algebra
+function +(I::InjMod, J::InjMod)
+  @req I.monoid_algebra == J.monoid_algebra "monoid algebras not the same"
+  return InjMod(I.monoid_algebra, vcat(I.indec_injectives, J.indec_injectives))
+end
+
+struct InjRes #ZZ^d-graded injective resolution
+  mod::SubquoModule
+  inj_mods::Vector{InjMod}
+  cochain_maps::Vector{MatElem}
+  upto::Int
+  Q_graded_part::IrrRes
+  shift::Vector{Int} #not needed
+end
+
+function Base.show(io::IO,J::InjMod)
+  print(
+    io, "injective module given by direct sum of ", length(J.indec_injectives)," indecomposable injectives"
+  )
+end
+
+function Base.show(io::IO, ::MIME"text/plain", J::InjMod)
+  println(io, "injective module given by direct sum of indecomposable injectives")
+  for Ji in J.indec_injectives
+    println(io, "  k{", Ji.vector, " + F - Q}, where p_F = ", Ji.face.prime)
+  end
+  print(
+    io,
+    "over ",
+    J.monoid_algebra
+  )
+end
+
+function Base.show(io::IO, ::MIME"text/plain", res::InjRes)
+  println(io, "injective resolution ")
+  println(io, "  ", join(["J^$i" for i in 0:res.upto], " -> "))
+  println(io, "where ")
+  for i in eachindex(res.inj_mods)
+    j = i-1
+    println(io, " J^$j = direct sum of")
+    for Ji in res.inj_mods[i].indec_injectives
+      println(io, "    k{", Ji.vector, " + F - Q}, where p_F = ", Ji.face.prime)
+    end
+  end
+  println(io, "of ", res.mod)
+  println(
+    io,
+    "over ",
+    base_ring(res.mod)
+  )
+end
+
+function Base.show(io::IO, ::MIME"text/plain", res::IrrRes)
+  println(io, "irreducible resolution ")
+  println(io, "  ", join(["W^$i" for i in 0:(length(res.irr_sums) - 1)], " -> "))
+  println(io, "where ")
+  for i in eachindex(res.irr_sums)
+    j = i-1
+    println(io, " W^$j = direct sum of")
+    for Ji in res.irr_sums[i].indec_injectives
+      println(io, "    k{", Ji.vector, " + F - Q}_Q, where p_F = ", Ji.face.prime)
+    end
+  end
+  println(io, "of ", res.mod)
+  println(
+    io,
+    "over ",
+    base_ring(res.mod)
+  )
+end
+
+function Base.show(io::IO, ::MIME"text/plain", Ji::IndecInj)
+  println(io, "indecomposable injective")
+  println(io, "  k{", Ji.vector, " + F - Q},")
+  print(io, "where p_F = ", Ji.face.prime)
+end
+
+function Base.show(io::IO, Ji::IndecInj)
+  print(
+      io, "indecomposable injective k{", Ji.vector, " + F - Q}, where p_F = ", Ji.face.prime
+    )
+end
+
+@doc raw"""
+  generators_W_H(kQ::MonoidAlgebra, H::HyperplaneQ, a::Vector{Int})
```

Comment:
Per convention these signature descriptions should be indented with 4 spaces -- this is because the docstring is parsed as Markdown, and the 4 space indent marks a code block. 
```suggestion
    generators_W_H(kQ::MonoidAlgebra, H::HyperplaneQ, a::Vector{Int})
```